terraform/eks/main.tf

```diff
@@ -469,14 +145,38 @@ module "validator" {
   region                       = var.region
   testing_id                   = module.common.testing_id
   metric_namespace             = "${module.common.otel_service_namespace}/${module.common.otel_service_name}"
-  sample_app_endpoint          = "http://${var.sample_app_mode == "push" ? kubernetes_service.sample_app_service[0].load_balancer_ingress.0.hostname : kubernetes_service.pull_mode_sample_app_service[0].load_balancer_ingress.0.hostname}:${module.common.sample_app_lb_port}"
-  mocked_server_validating_url = "http://${kubernetes_service.mocked_server_service.load_balancer_ingress.0.hostname}/check-data"
-
+  sample_app_endpoint          = length(kubernetes_service.sample_app_service) > 0 ? "http://${kubernetes_service.sample_app_service.0.load_balancer_ingress.0.hostname}:${module.common.sample_app_lb_port}" : ""
+  mocked_server_validating_url = length(kubernetes_service.mocked_server_service) > 0 ? "http://${kubernetes_service.mocked_server_service.0.load_balancer_ingress.0.hostname}/check-data" : ""
+  cloudwatch_context_json = var.aoc_base == "prometheus" ? jsonencode({
```

Comment:
now i kinda understand what's `aoc_base`.  this name seems not quite verbose, and in some cases like testing promotheus exporter we uses mock server as well.  maybe think of another name? for example: `launch_mock_server` ?

if the only difference is mockserver, maybe we just leave mockserver there so that we don't need those ifelses?

Reply:
Thanks for the second thought. Another difference is declared here. Let me know if the current name would look any better. https://github.com/aws-observability/aws-otel-test-framework/pull/280#discussion_r624341233